cmake/FindLibuv.cmake

```diff
@@ -25,8 +25,13 @@ if(NOT WIN32)
   set(_LIBUV_ROOT_PATHS "/usr/"
                         "/usr/local/")
   if(_LIBUV_FOUND)
-    set(_LIBUV_ROOT_PATHS ${_LIBUV_ROOT_PATHS}
-                          ${_LIBUV_LIBDIR})
+    if(APPLE)
+      set(_LIBUV_ROOT_PATHS ${_LIBUV_ROOT_PATHS}
+              ${_LIBUV_PREFIX})
```

Comment:
on macos uv.h ends up in `/opt/homebrew/Cellar/libuv/1.46.0/include/uv.h`, but `_LIBUV_LIBDIR` resolves to `/opt/homebrew/Cellar/libuv/1.46.0/lib`

Reply:
You should be able to address this by specifying the libuv root dir directly.  Try the following to see if it works:

```
LIBUV_ROOT_DIR=/opt/homebrew/Cellar/libuv/1.46.0 cmake ..
```

Reply:
Thanks, that worked. I've changed this patch to update the building documentation to reflect that.

Reply:
Another possible solve is to have it search in `/opt/homebrew` when building on macOS which means it compiles without having to set additional env vars and keeps the search code common across all architectures (since we're just adding an extra common location). If this is still sketchy I can revert that and leave it as a comment in the docs.

Reply:
Oh, nice!  So adding /opt/homebrew to _LIBUV_ROOT_PATHS in the Apple case does make the build work without anything else?  I don't have a problem with that change at all; we actually do [something very similar to this](https://github.com/datastax/cpp-driver/blob/master/cmake/FindOpenSSL.cmake#L87) on the OpenSSL side.  /usr/local/opt/openssl isn't obviously an OSX path but it's not something you see on your average Linux system... and it does appear to be the spot where the Macs install OSX.

I don't think adding /opt/homebrew in the OSX case is janky at all.  If this fixes enables libuv detection automatically I'd say let's go with it.

Reply:
Confirmed directly with @hhughes that everything builds correctly on OSX with just the _LIBUV_ROOT_PATHS change above (as in there's no additional steps required to handle the libuv discovery/include bits).  Calling that good.